pkg/cmd/dedicated/register/registercluster.go

```diff
@@ -0,0 +1,350 @@
+
+	"github.com/redhat-developer/app-services-cli/pkg/shared/factory"
+	"github.com/spf13/cobra"
+	"strings"
+)
+
+type options struct {
+	selectedClusterId string
+	clusterList       v1.ClusterList
+	//interactive                   bool
+	selectedCluster         v1.Cluster
+	clusterMachinePoolList  v1.MachinePoolList
+	existingMachinePoolList []v1.MachinePool
+	//validatedMachinePoolList      []v1.MachinePool
+	selectedClusterMachinePool    v1.MachinePool
+	requestedMachinePoolNodeCount int
+	accessKafkasViaPrivateNetwork bool
+	newMachinePool                v1.MachinePool
+
+	f *factory.Factory
+}
+
+const (
+	machinePoolId           = "kafka-standard"
+	machinePoolTaintKey     = "bf2.org/kafkaInstanceProfileType"
+	machinePoolTaintEffect  = "NoExecute"
+	machinePoolTaintValue   = "standard"
+	machinePoolInstanceType = "m5.2xlarge"
+	machinePoolLabelKey     = "bf2.org/kafkaInstanceProfileType"
+	machinePoolLabelValue   = "standard"
+)
+
+func NewRegisterClusterCommand(f *factory.Factory) *cobra.Command {
+	opts := &options{
+		f: f,
+	}
+
+	//TODO add Localizer
+	cmd := &cobra.Command{
+		Use: "register-cluster",
+		// Short: f.Localizer.MustLocalize("registerCluster.cmd.shortDescription"),
+		Short:   "registerCluster.cmd.shortDescription",
+		Long:    "registerCluster.cmd.longDescription",
+		Example: "registerCluster.cmd.example",
+		Args:    cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			return runRegisterClusterCmd(opts)
+		},
+	}
+
+	// TODO add Localizer and logic for pass cluster ID flag
+	flags := kafkaFlagutil.NewFlagSet(cmd, f.Localizer)
+	// this flag will allow the user to pass the cluster id as a flag
+	opts.selectedClusterId = "123"
+	flags.StringVar(&opts.selectedClusterId, "cluster-id", "", "cluster id")
+	//flags.StringVar(&opts.selectedClusterId, "cluster-id", "", f.Localizer.MustLocalize("registerCluster.flag.clusterId"))
+
+	return cmd
+}
+
+func runRegisterClusterCmd(opts *options) error {
+	getListClusters(opts)
+	runClusterSelectionInteractivePrompt(opts)
+	getMachinePoolList(opts)
+	selectAccessPrivateNetworkInteractivePrompt(opts)
+	registerClusterWithKFM(opts)
+
+	return nil
+}
+
+func getListClusters(opts *options) error {
+	// ocm client connection
+	conn, err := opts.f.Connection()
+	if err != nil {
+		return err
+	}
+	client, err := conn.API().OCMClustermgmt()
+	if err != nil {
+		return err
+	}
+
+	resource := client.Clusters().List()
+	response, err := resource.Send()
+	if err != nil {
+		return err
+	}
+	clusters := response.Items()
+	opts.clusterList = *clusters
+	return nil
+}
+
+func runClusterSelectionInteractivePrompt(opts *options) error {
+	clusterListString := make([]string, 0)
+	for _, cluster := range opts.clusterList.Slice() {
+		clusterListString = append(clusterListString, cluster.Name())
+	}
+
+	// TODO add page size and Localizer
+	prompt := &survey.Select{
+		Message: "Select the cluster to register",
```

Comment:
```suggestion
		Message: "Select the ready cluster to register",
```

related to https://github.com/redhat-developer/app-services-cli/pull/1795/files#r1084114798